code/modules/projectiles/updated_projectiles/ammunition.dm

```diff
@@ -355,6 +355,165 @@ Turn() or Shift() as there is virtually no overhead. ~N
 
 
 
+//Deployable ammo box
+
+/obj/item/ammobox
+	name = "M41A Ammo Box"
+	desc = "A large, deployable ammo box."
+	w_class = 5
+	icon = 'icons/obj/items/ammo.dmi'
+	icon_state = "ammobox"
+	item_state = "ammobox"
+	var/base_icon_state = "ammobox"
+	var/magazine_amount = 10
+	var/max_magazine_amount = 10
+	var/max_magazine_rounds = 40
+	var/ammo_type = /datum/ammo/bullet/rifle
+	var/magazine_type = /obj/item/ammo_magazine/rifle
+	var/deployed = 0
+	var/base = /obj/item/ammobox
+
+/obj/item/ammobox/update_icon()
+	if(magazine_amount > 0) 
+		icon_state = "[base_icon_state]_deployed"
+	else 
+		icon_state = "[base_icon_state]_empty"
+
+/obj/item/ammobox/examine(mob/user)
+	to_chat(user, "It contains [magazine_amount] out of [max_magazine_amount] magazines\s.")
+
+/obj/item/ammobox/attackby(obj/item/I, mob/user)
+	if(deployed == 0)
+		to_chat(user, "<span class='warning'>[src] must be on the ground to be refilled.</span>")
+		return
+	if(magazine_amount == max_magazine_amount)
+		to_chat(user, "<span class='warning'>The [src] is already full.")
```

Comment:
This message should happen if you try to restock ammo. For other objects it should just `return`